### DISCUSSION TITLE
Can you publish guides about how can use it

### DISCUSSION THREAD

**Discussion by Ravid-Levy** [cmusphinx/pocketsphinx#361] [Q&A]
Can you publish guides about how can use it in java language or other programming language instead of python?

And can you publish what are the changes that you are made in this repo?

Thanks
Ravid

**Comment by dhdaines** [comment]
No, because the Java interface code was unmaintained, I don't actually know how to use it, there wasn't any existing documentation, and I don't have time to learn in order to maintain it.  If someone else wants to do so, they are welcome!

**Comment by smbika007** [comment]
Hi, can you tell me how to run pocketsphinx_main to use a microphone? I tried a command line with arguments "single test.wav" and language model files and it returned output indicating that it did process the file. I substituted "live" for "single test.wav" and it immediately errored out with the Usage bolierplate. If I am supposed to use the "live.exe" app the solution has in it, I can't. It was the only thing that did not build. It had unresolved references. Could that be because I had to generate solution using CMake with the INSTALL_GRSTREAM option unchecked? When checked, CMake returns an unresolved reference error to that.

I come from having successfully implemented code using the SourceForge 5prealpha repos and was hopeful this newer github version would work pretty much the same way.

**Comment by dhdaines** [comment]
First, ask yourself if you *really* want to run a command-line program on Windows to do live speech recognition from the microphone.  Is this a useful thing for you to do?  Tell me your use case and I will let you know how to do it.

**Comment by yuvalkry** [comment]
In the 5prealpha version the MSVC solution created a DLL and there was SWIG code for interfacing with it.
With the new foreign API of Java, it would be possible to call the DLL functions more easily without SWIG.

But - the current cmake file does not contain the instructions for creating a DLL. Could you please add it?
  Thanks!
